packages/url_launcher/url_launcher/lib/src/url_launcher_uri.dart

```diff
@@ -11,25 +11,13 @@ import 'type_conversion.dart';
 
 /// Passes [url] to the underlying platform for handling.
 ///
-/// [mode] support varies significantly by platform:
-///   - [LaunchMode.platformDefault] is supported on all platforms:
-///     - On iOS and Android, this treats web URLs as
-///       [LaunchMode.inAppWebView], and all other URLs as
-///       [LaunchMode.externalApplication].
-///     - On Windows, macOS, and Linux this behaves like
-///       [LaunchMode.externalApplication].
-///     - On web, this uses `webOnlyWindowName` for web URLs, and behaves like
-///       [LaunchMode.externalApplication] for any other content.
-///   - [LaunchMode.inAppWebView] is currently only supported on iOS and
-///     Android. If a non-web URL is passed with this mode, an [ArgumentError]
-///     will be thrown.
-///   - [LaunchMode.externalApplication] is supported on all platforms.
-///     On iOS, this should be used in cases where sharing the cookies of the
-///     user's browser is important, such as SSO flows, since Safari View
-///     Controller does not share the browser's context.
-///   - [LaunchMode.externalNonBrowserApplication] is supported on iOS 10+.
-///     This setting is used to require universal links to open in a non-browser
-///     application.
+/// [mode] support varies significantly by platform. Clients can use
+/// [supportsLaunchMode] to query for support, but platforms will fall back to
+/// other modes if the requested mode is not supported, so checking is not
+/// required. The default behavior of [LaunchMode.platformDefault] is up to each
+/// platform, and its behavior for a given platform may change over time as new
+/// modes are supported, so clients that want a specific mode should request it
+/// rather than rely on any currently observed default behavior.
```

Comment:
This makes the documentation federation-friendly and evergreen. If we find that people want docs about specific platform behaviors, we can add it to the platform package READMEs later.

Reply:
Per-platform docs are the way, agreed!